### PR TITLE
Move build to root folder, include both `dumper` and `client` tools 

### PR DESCRIPTION
**Pull request by paolomorandini**
The intention is to have the same commands perform the same actions whether they are executed in the repository top-level folder or in the unzipped archive top-level folder

Test archive:
```
./gradlew :publishedDistZip
unzip build/distributions/dwh-migration-tools.zip
cd dwh-migration-tools
bin/dwh-migration-dumper --help # invoke with meaningful args
bin/dwh-migration-client --help # invoke with meaningful args
```

Test repository:
```
bin/dwh-migration-dumper --help # invoke with meaningful args
bin/dwh-migration-client --help # invoke with meaningful args
```

Both are expected to have the same behavior.

### CODE REVIEW

File: dumper/lib-dumper-spi/build.gradle [hunk 1]
```diff
@@ -19,6 +19,6 @@ plugins {
 }
 
 dependencies {
-	api "com.fasterxml.jackson.core:jackson-databind"
+	implementation "com.fasterxml.jackson.core:jackson-databind"
```
Comment (paolomorandini) [inline dumper/lib-dumper-spi/build.gradle:22]: Gradle wouldn't pick the pinned versions in `buildSrc/.../conventions` for `api`

**Review by bdw-g — CHANGES_REQUESTED** [review]
Looks good, but, when I run this:
```
git clean -xdf
./bin/dwh-migration-dumper
```

Then I get the following:

```
wiegmans@nifftuk:~/Code/google/dwh-migration-tools$ ./bin/dwh-migration-dumper
+++ readlink -f ./bin/dwh-migration-dumper
++ dirname /usr/local/google/home/wiegmans/Code/google/dwh-migration-tools/bin/dwh-migration-dumper
+ BIN=/usr/local/google/home/wiegmans/Code/google/dwh-migration-tools/bin/../dumper/app/build/install/app/bin/dwh-migration-dumper
+ '[' '!' -x /usr/local/google/home/wiegmans/Code/google/dwh-migration-tools/bin/../dumper/app/build/install/app/bin/dwh-migration-dumper ']'
+ cd dumper
+ ./gradlew --parallel :app:installDist
./bin/dwh-migration-dumper: line 6: ./gradlew: No such file or directory
```

So there may be some further updates required

**Review by bdw-g — APPROVED** [review]
tested it and it works

**Review by genemerrill — APPROVED** [review]
works for me too

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+            from("client") {
+                exclude "**/__pycache__"
+                into "client"
+            }
+            from("bin/dwh-migration-client") {
+                into "bin"
+            }
+            project(":dumper:app").afterEvaluate {
+                from it.tasks.installPublishedDist
+            }
```
Comment (shevek-google) [inline build.gradle:18]: This will do, but for the future, I'm a bit nervous about an afterEvaluate.

**Pull request merged**